q2_dummy_types/_int_sequence.py

```diff
@@ -30,66 +29,60 @@
+# Define a file format for use in the directory format defined below.
+class IntSequenceFormat(model.TextFileFormat):
+    # Sniffers are used when reading and writing data from a file. A sniffer
     # determines whether a file appears to conform to its file format. Sniffers
     # should only read a small piece of the file to determine membership and
     # should not rely on file extension.
-    @classmethod
-    def sniff(cls, filepath):
-        with open(filepath, 'r') as fh:
+    def sniff(self):
+        with self.open() as fh:
             for line, _ in zip(fh, range(5)):
                 try:
                     int(line.rstrip('\n'))
                 except (TypeError, ValueError):
                     return False
             return True
 
-# Define a data layout with name 'int-sequence' and version 1. A data layout is
-# a directory structure composed of one or more files (nested directories are
-# also supported). Each file has a specific file format associated with it.
-# This data layout only has a single file, ints.txt, with file format
-# `IntSequenceFormat`.
-int_sequence_data_layout = qiime.plugin.DataLayout('int-sequence', 1)
-int_sequence_data_layout.register_file('ints.txt', IntSequenceFormat)
-
-# Register the data layout on the plugin.
-plugin.register_data_layout(int_sequence_data_layout)
-
-# Register the data layout with the semantic types defined above. A data layout
-# can be registered to multiple semantic types. Currently, a semantic type can
-# only have a single data layout associated with it.
-plugin.register_type_to_data_layout(IntSequence1, 'int-sequence', 1)
-plugin.register_type_to_data_layout(IntSequence2, 'int-sequence', 1)
-
-
-# Define a data layout reader for reading the data layout into a view type
-# (`list` in this case). Any function name can be used but the convention is
-# `<data-layout-name>_to_<view-type>`.
-def int_sequence_to_list(data_dir):
-    with open(os.path.join(data_dir, 'ints.txt'), 'r') as fh:
-        view = []
+# Define a directory format. A directory format is a directory structure
+# composed of one or more files (nested directories are also supported). Each
+# file has a specific file format associated with it.  This directory format
+# only has a single file, ints.txt, with file format `IntSequenceFormat`.
+IntSequenceDirectoryFormat = model.SingleFileDirectoryFormat(
+    'IntSequenceDirectoryFormat', 'ints.txt', IntSequenceFormat)
+
+# Register the directory format with the semantic types defined above. A
+# directory format can be registered to multiple semantic types. Currently, a
+# semantic type can only have a single directory format associated with it.
+plugin.register_semantic_type_to_format(
+    IntSequence1,
+    artifact_format=IntSequenceDirectoryFormat)
+plugin.register_semantic_type_to_format(
+    IntSequence2,
+    artifact_format=IntSequenceDirectoryFormat)
+
+
+# Define a transformer for reading a file format (`IntSequenceFormat`) into a
```

Comment:
`for converting` might be better. Reading/Writing are probably more confusing than helpful for transformers